doc/languages-frameworks/coq.section.md

```diff
@@ -10,6 +10,35 @@ The Coq derivation is overridable through the `coq.override overrides`, where ov
 
 The associated package set can be obtained using `mkCoqPackages coq`, where `coq` is the derivation to use.
 
+## Creating custom Coq environments with `coq.withPackages` {#coq-withPackages}
+
+The `coq.withPackages` function provides a convenient way to create a Coq environment that includes additional Coq packages. This is similar to how `python.withPackages` works for Python environments.
+
+The function takes a function that receives the Coq package set and returns a list of packages. It returns a wrapped Coq environment where all Coq binaries (`coqtop`, `coqc`, `coqdep`, `coqchk`, `coqide`, etc.) are configured with the appropriate environment variables to find the packages.
+
+### Usage {#coq-withPackages-usage}
+
+Here is an example of creating a Coq environment with specific packages.
+
+```nix
+coq.withPackages (
+  ps: with ps; [
+    mathcomp
+    bignums
+  ]
+)
+```
+
+For more flexibility, you can use `coq.withPackages'` with a list (which can make use of arbitrary package sets).
+
+
+```nix
+coq_8_20.withPackages' [
+  coqPackages_8_19.mathcomp
```

Comment:
This example is a bit strange, there is no good reason for this to work.

Reply:
Okay, is there a better example perhaps I can include for using `withPackages'`? Or is it not useful at all?